src/conn/conn_dhandle.c

```diff
@@ -716,7 +716,8 @@ __wt_conn_btree_apply(WT_SESSION_IMPL *session, const char *uri,
                 goto done;
 
             if (!F_ISSET(dhandle, WT_DHANDLE_OPEN) || F_ISSET(dhandle, WT_DHANDLE_DEAD) ||
-              !WT_DHANDLE_BTREE(dhandle) || dhandle->checkpoint != NULL || WT_IS_METADATA(dhandle))
+              !WT_DHANDLE_BTREE(dhandle) || dhandle->checkpoint != NULL ||
+              WT_IS_METADATA(dhandle) || WT_SUFFIX_MATCH(dhandle->name, ".wtobj"))
```

Comment:
I assume this new test here is needed to handle the local file. Because that is a `file:` entry it probably generates a `WT_DHANDLE_TYPE_BTREE` handle. But because it's a piece of a tiered table it isn't, in fact, a btree.

So maybe the `.wtobj` test should be part of `WT_DHANDLE_BTREE()`? That would take a bit of homework/testing to see if anything breaks. But I'm guessing that any other code that tries to do something meaningful with the dhandle on the local file would have failed similar to the way this issue manifested.

Reply:
If you want we can open a new ticket about whether testing .wtobj should be part of WT_DHANDLE_BTREE. That broader and more invasive so should be considered on its own.

Reply:
I created WT-11562 for the follow-up investigation.